Supervisor/model.c

```diff
@@ -0,0 +1,83 @@
+#include <math.h>   // for fabs()
+#include "config.h"
+#include "model.h"
+
+void setStartValues(ModelInstance *comp) {
+    M(s) = false;       // Clock
+    M(x) = 0.0;         // Sample
+    M(as) = 1.0;        // Discrete state/output
+    M(stateEvent) = false;
+}
+
+Status calculateValues(ModelInstance *comp) {
+    UNUSED(comp);
+    return OK;
+}
+
+Status getFloat64(ModelInstance* comp, ValueReference vr, double *value, size_t *index) {
+    switch (vr) {
+        case vr_as:
+            if (comp->state == EventMode && M(stateEvent)) {
+                value[(*index)++] = M(as)*-1.0;
+            }
+            else {
+                value[(*index)++] = M(as);
+            }
+            return OK;
+        default:
+            logError(comp, "Unexpected value reference: %d.", vr);
+            return Error;
+    }
+}
+
+Status setFloat64(ModelInstance* comp, ValueReference vr, const double *value, size_t *index) {
+    switch (vr) {
+        case vr_x:
+            M(x) = value[(*index)++];
+            return OK;
+        default:
+            logError(comp, "Unexpected value reference: %d.", vr);
+            return Error;
+    }
+}
+
+Status getClock(ModelInstance* comp, ValueReference vr, bool* value) {
+    switch (vr) {
+        case vr_s:
+            if (comp->state == EventMode && M(stateEvent)) {
+                (*value) = true;
+            }
+            else {
+                (*value) = true;
+            }
+            return OK;
+        default:
+            logError(comp, "Unexpected value reference: %d.", vr);
+            return Error;
+    }
+}
+
+void enterEventMode(ModelInstance* comp, int stepEvent, int stateEvent, const int rootsFound[], size_t nEventIndicators, int timeEvent) {
+    UNUSED(stepEvent);
+    UNUSED(rootsFound);
+    UNUSED(nEventIndicators);
+    UNUSED(timeEvent);
+
+    M(stateEvent) = stateEvent == 1;
```

Comment:
This is not robust if `stateEvent == 2` (fmi3EventUnknown).

Reply:
Hi @t-sommer @clagms 
I cannot (or I do not know) how to retrieve the FMUs from here. Would it be possible to zip them and share it here?
Then I could test them on my side.
Thanks,
Masoud

Reply:
You can just [download the this branch](https://github.com/t-sommer/Reference-FMUs/archive/refs/heads/synchronous-clocks.zip) and follow the instructions in the [readme](https://github.com/modelica/Reference-FMUs#build-the-fmus).